zerver/lib/bugdown/__init__.py

```diff
@@ -94,27 +94,20 @@ def rewrite_local_links_to_relative(link: str) -> str:
 
     return link
 
-def url_embed_preview_enabled_for_realm(message: Optional[Message]) -> bool:
+def url_embed_preview_enabled_for_realm(message: Optional[Message]=None) -> bool:
     if message is not None:
-        realm = message.get_realm()  # type: Optional[Realm]
+        realm = message.get_realm()
     else:
-        realm = None
+        realm = arguments.current_realm
 
     if not settings.INLINE_URL_EMBED_PREVIEW:
         return False
-    if realm is None:
-        return True
     return realm.inline_url_embed_preview
 
 def image_preview_enabled_for_realm() -> bool:
-    if arguments.current_message is not None:
-        realm = arguments.current_message.get_realm()  # type: Optional[Realm]
-    else:
-        realm = None
+    realm = arguments.current_realm
     if not settings.INLINE_IMAGE_PREVIEW:
         return False
-    if realm is None:
-        return True
```

Comment:
I don't think we can remove these, because `arguments.current_realm` comes from `do_convert`, and `message_realm` there can be `None` (at least as of this commit).  I think we mostly use that code path in tests, though.

Reply:
Ok, I fixed this.

Background:

So  `realm` can indeed be `None` when we generate docs.  This seems pretty brittle, but it should only be a pitfall for odd use cases, hopefully.  We'll set `realm` as long as `message` is passed in, but amazingly even `message` is optional, since we really only need `content`.  As i mentioned on chat, the `message` object is really only there to piggyback things like mentioned user ids.  But we should eventually make it a required parameter and let the documentation code pass in a `FakeMessage` object (or figure out a better scheme for returning out-of-band data).

And we should probably just have the calling code compute all flags related to the `Realm` object, so we're less coupled to the DB model within in the parser.  (Instead of lazily looking up these things, we should just pre-compute them, so that we fail fast rather than deep in the innards.)